docs/experiments/Data_Set.ipynb

```diff
@@ -0,0 +1,490 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, we load the data set."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data, test_data = load()\n",
+    "X = np.concatenate([train_data[0], test_data[0]])\n",
+    "Y = np.concatenate([train_data[2], test_data[2]])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Then, we sum over the 12 binary inputs of $X$ and determine the share of $Y=1$ per each possible sum value. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Share of Y=1 per each value of the sum of the binary inputs of X:\n",
+      "0 :  0.0 %\n",
+      "1 :  0.0 %\n",
+      "2 :  0.0 %\n",
+      "3 :  0.0 %\n",
+      "4 :  0.0 %\n",
+      "5 :  7.575757575757576 %\n",
+      "6 :  57.57575757575758 %\n",
+      "7 :  92.42424242424242 %\n",
+      "8 :  100.0 %\n",
+      "9 :  100.0 %\n",
+      "10 :  100.0 %\n",
+      "11 :  100.0 %\n",
+      "12 :  100.0 %\n"
+     ]
+    }
+   ],
+   "source": [
+    "X_sum_Y = []\n",
+    "\n",
+    "for i in range(13):\n",
+    "    X_sum_Y.append([])\n",
+    "\n",
+    "for i in range(len(X)):\n",
+    "    j = np.sum(X[i])\n",
+    "    X_sum_Y[j].append(Y[i])\n",
+    "\n",
+    "print('Share of Y=1 per each value of the sum of the binary inputs of X:')\n",
+    "for i in range(13):\n",
+    "    print(i, ': ', 100 * np.sum(np.array(X_sum_Y[i]))/len(X_sum_Y[i]), '%')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Our attempt to generate the data set above"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "After analyzing the provided data set, we try to generate a similar one ourselves since the explicit algorithm for doing so was not provided in the paper.\n",
+    "\n",
+    "The first task is to sample 12 uniformly distributed points on a unit 2-sphere. For that purpose, we randomly sample $\\theta$ from $[0, 2\\pi]$ using the uniform distribution. Then, we randomly sample a value for the cosine of $\\phi$ from $[-1, 1]$ again using the uniform distribution. Finally, we apply arccosine to the sampled cosine of $\\phi$ to get $\\phi$ itself. Contrary to the convention, we have assigned $\\theta$ to the aziumthal angle and $\\phi$ to the polar angle to keep the naming consistent with SciPy's spherical harmonics implementation.\n",
+    "\n",
+    "Alternatively, one can randomly sample a 3-dimensional vector from $\\mathbb{R}^3$ using standard normal distribution, normalize it, and convert Cartesian coordinates to spherical coordinates."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
```

Comment:
numpy is repeatedly imported, we could keep all import in the first cell of the notebook